spring-integration-core/src/main/java/org/springframework/integration/core/ErrorMessagePublisher.java

```diff
@@ -70,7 +70,7 @@ public final void setErrorMessageStrategy(ErrorMessageStrategy errorMessageStrat
 		this.errorMessageStrategy = errorMessageStrategy;
 	}
 
-	public final void setChannel(MessageChannel channel) {
+	public final void setChannel(@Nullable MessageChannel channel) {
```

Comment:
And why is this, please?
Why we cannot keep contract as non-null, but preserve nullability internally?